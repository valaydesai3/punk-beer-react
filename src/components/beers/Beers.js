import React, { useContext } from 'react';
import BeerItem from './BeerItem';
import Spinner from '../layout/Spinner';
import PunkBeerContext from '../../context/punkBeerContext';

const Beers = () => {
  const punkBeerContext = useContext(PunkBeerContext);
  const {
    searchBeerLoading,
    searchBeerFlag,
    searchText,
    beers
  } = punkBeerContext;

  if (searchBeerLoading) {
    return <Spinner />;
  }

  if (searchBeerFlag && beers.length === 0) {
    return <h3 data-testid="noResults">No beers found for "{searchText}"</h3>;
  } else if (searchBeerFlag && beers.length > 0) {
    return (
      <div data-testid="searchResults" style={beerStyle}>
        <h3>Search Results for "{searchText}"</h3>
        {beers.map(beer => (
          <BeerItem key={beer.id} beer={beer} />
        ))}
      </div>
    );
  } else {
    return null;
  }
};

const beerStyle = {};

export default Beers;
