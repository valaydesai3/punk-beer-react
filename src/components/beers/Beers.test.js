// __tests__/fetch.test.js
import React from 'react';
import { render, cleanup } from '@testing-library/react';
import 'jest-dom/extend-expect';
import axiosMock from 'axios';
import PunkBeerContext from '../../context/punkBeerContext';
import Beers from './Beers';

afterEach(cleanup);

const beers = [
  {
    id: 1,
    name: 'Chaos Theory',
    tagline: 'Predictably Random IPA.',
    first_brewed: '10/2008',
    description:
      'Chaos Theory is the most under- rated achievement of 20th Century science. This beer can only aspire to parallel to the mathematical use of the word chaos, which is at odds to the common parlance. The purest showcase of the magnificent hop that is Nelson Sauvin; grapefruit, pineapple and caramel sing above the chaos of life.',
    image_url: 'https://images.punkapi.com/v2/124.png',
    abv: 7.1
  },
  {
    id: 2,
    name: 'Test beer',
    tagline: 'Test beer tagline',
    first_brewed: '10/2008',
    description: 'Test beer desc',
    image_url: 'https://images.punkapi.com/v2/124.png',
    abv: 4.1
  }
];

test('renders beers component with no beers found', async () => {
  const { getByTestId } = render(
    <PunkBeerContext.Provider
      value={{
        searchBeerLoading: false,
        searchBeerFlag: true,
        searchText: 'test',
        beers: []
      }}
    >
      <Beers />
    </PunkBeerContext.Provider>
  );
  expect(getByTestId('noResults')).toHaveTextContent(
    'No beers found for "test"'
  );
  expect(getByTestId('noResults')).toMatchSnapshot();
});

test('renders beers component with beers found', async () => {
  const { getByTestId } = render(
    <PunkBeerContext.Provider
      value={{
        searchBeerLoading: false,
        searchBeerFlag: true,
        searchText: 'test',
        beers: beers
      }}
    >
      <Beers />
    </PunkBeerContext.Provider>
  );
  expect(getByTestId('searchResults')).toHaveTextContent('Chaos Theory');
  expect(getByTestId('searchResults')).toMatchSnapshot();
});
