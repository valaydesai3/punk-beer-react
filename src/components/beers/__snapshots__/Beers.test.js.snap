// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders beers component with beers found 1`] = `
<div
  data-testid="searchResults"
>
  <h3>
    Search Results for "
    test
    "
  </h3>
  <div
    class="card beer-item"
    data-testid="beerItem"
  >
    <div
      class="card-grid-img"
    >
      <img
        alt="Beer"
        src="https://images.punkapi.com/v2/124.png"
      />
    </div>
    <div
      class="card-grid-body"
    >
      <p
        class="card-grid-name"
      >
        Chaos Theory
      </p>
      <p
        class="card-grid-description"
      >
        Chaos Theory is the most under- rated achievement of 20th Century science. This beer can only aspire to parallel to the mathematical use of the word chaos, which is at odds to the common parlance. The purest showcase of the magnificent hop that is Nelson Sauvin; grapefruit, pineapple and caramel sing above the chaos of life.
      </p>
      <p
        class="card-grid-tagline"
      >
        Predictably Random IPA.
        <span>
          10/2008
        </span>
        <span>
          7.1
          %
        </span>
      </p>
    </div>
  </div>
  <div
    class="card beer-item"
    data-testid="beerItem"
  >
    <div
      class="card-grid-img"
    >
      <img
        alt="Beer"
        src="https://images.punkapi.com/v2/124.png"
      />
    </div>
    <div
      class="card-grid-body"
    >
      <p
        class="card-grid-name"
      >
        Test beer
      </p>
      <p
        class="card-grid-description"
      >
        Test beer desc
      </p>
      <p
        class="card-grid-tagline"
      >
        Test beer tagline
        <span>
          10/2008
        </span>
        <span>
          4.1
          %
        </span>
      </p>
    </div>
  </div>
</div>
`;

exports[`renders beers component with no beers found 1`] = `
<h3
  data-testid="noResults"
>
  No beers found for "
  test
  "
</h3>
`;
